std/algorithm/searching.d

```diff
@@ -1245,45 +1245,90 @@ private auto extremum(alias map = "a", alias selector = "a < b", Range,
     if (isInputRange!Range && !isInfinite!Range &&
         !is(CommonType!(ElementType!Range, RangeElementType) == void))
 {
-    alias mapFun = unaryFun!map;
-    alias selectorFun = binaryFun!selector;
+    enum isMappingFirst = __traits(compiles, unaryFun!map(seedElement));
+
+    // check for identity ("a")
+    static if (isSomeString!(typeof(map)))
+        enum isIdentity = map == "a";
```

Comment:
It would be nicer to specialize on no mapping function. Would that require a lot more code?